apps/web/src/app/[locale]/tracks/[slug]/page.tsx

```diff
@@ -10,7 +13,13 @@ interface Props {
 }
 
 // todo: write a suspense skeleton...
-export default function Page({ params }: Props) {
+export default async function Page({ params }: Props) {
+  const session = await getServerAuthSession();
+  const flags = await getAllFlags();
+
+  if (!session && flags.enableEarlyAccess) {
+    return redirect('/waitlist');
+  }
```

Comment:
We could extract this logic in a hook instead of duplicating it in many places, though I think the `redirect` will have to be duplicated

Reply:
eh actually ended up being kinda annoying to do, gonna ship it and strip it later